Runtime/JSValue.cs

```diff
@@ -9,30 +9,28 @@ namespace NodeApi;
 public readonly struct JSValue
 {
     private readonly napi_value _handle;
+    private readonly JSValueScope? _scope;
 
-    public JSValueScope Scope { get; }
+    public readonly JSValueScope Scope =>
+        _scope ?? JSValueScope.Current ?? throw new InvalidOperationException("No current scope");
 
-    public JSValue(JSValueScope scope, napi_value handle)
+    public JSValue() { }
+
+    public JSValue(napi_value handle) : this(JSValueScope.Current, handle)
     {
-        if (handle.Handle == nint.Zero)
-        {
-            throw new ArgumentException($"{nameof(handle)} must not be null.");
-        }
-        Scope = scope;
-        _handle = handle;
     }
 
-    public JSValue(napi_value handle)
+    public JSValue(JSValueScope? scope, napi_value handle)
     {
-        if (handle.Handle == nint.Zero)
+        if (handle.Handle != nint.Zero)
         {
-            throw new ArgumentException($"{nameof(handle)} must not be null.");
+            ArgumentNullException.ThrowIfNull(nameof(scope));
```

Comment:
`nameof(scope)` will never be null - delete `nameof()`.

Reply:
Is it valid to have a 0 handle and non-null scope?

Reply:
Yes, it is valid - it will return Undefined in this case too.